jobtastic/tests/__init__.py

```diff
@@ -0,0 +1,17 @@
+from contextlib import contextmanager
+import celery
+import mock
+
+
+@contextmanager
+def allow_checking_status_for_eager(*args, **kwargs):
+    if not celery.VERSION[0] == 4:
+        yield
+    else:
+        # In celery 4.x you cannot retrieve the status of a task if it's eager.
+        # This gets around that.
+        with mock.patch(
+            'celery.backends.base.Backend._ensure_not_eager',
+            autospec=True,
+        ):
+            yield
```

Comment:
im not really sure what is going on here.
i mean i understand that it's patching `celery.backends.base.Backend._ensure_not_eager` but what does that have to do with `assert meta.status == PROGRESS`?

Reply:
The comment above is supposed to be explaining it. But in celery4, you cannot check the status of an eager task. So if we want to check it anyway, we need to skip the check ensuring that the task isn't eager.

Reply:
ah so this patching, makes celery think the task is not eager?

Reply:
Pretty much, yea.